integration-tests/smoke/ccip/ccip_test.go

```diff
@@ -206,3 +208,797 @@ func TestTokenTransfer(t *testing.T) {
+
+		// get the header for the destination chain and the relevant block number
+		latesthdr, err := destChain.Client.HeaderByNumber(testcontext.Get(t), nil)
+		require.NoError(t, err)
+		block := latesthdr.Number.Uint64()
+		startBlocks[dest] = &block
+
+		// Send to the receiver on the destination chain paying with LINK token
+		var (
+			receiver = common.LeftPadBytes(state.Chains[dest].Receiver.Address().Bytes(), 32)
+			data     = []byte("")
+			feeToken = linkToken.Address()
+		)
+
+		// Increase the token send amount to more than available to intentionally cause a revert
+		tokens[src][0].Amount = new(big.Int).Mul(twoCoins, big.NewInt(100))
+
+		ccipMessage := router.ClientEVM2AnyMessage{
+			Receiver:     receiver,
+			Data:         data,
+			TokenAmounts: tokens[src],
+			FeeToken:     feeToken,
+			ExtraArgs:    nil,
+		}
+
+		_, _, err = changeset.CCIPSendRequest(
+			e,
+			state,
+			src, dest,
+			true,
+			ccipMessage,
+		)
+
+		require.Error(t, err)
+
+	})
+}
+
+func Test_PricingForMessages(t *testing.T) {
+	t.Parallel()
+	lggr := logger.TestLogger(t)
+	// ctx := changeset.Context(t)
+	tenv, _, _ := testsetups.NewLocalDevEnvironmentWithDefaultPrice(t, lggr, nil)
+	e := tenv.Env
+
+	// feeds := state.Chains[tenv.FeedChainSel].USDFeeds
+	output, err := changeset.DeployPrerequisites(tenv.Env, changeset.DeployPrerequisiteConfig{
+		ChainSelectors: tenv.Env.AllChainSelectors(),
+	})
+	require.NoError(t, err)
+	require.NoError(t, tenv.Env.ExistingAddresses.Merge(output.AddressBook))
+
+	expectedSeqNum := make(map[changeset.SourceDestPair]uint64)
+	expectedSeqNumExec := make(map[changeset.SourceDestPair][]uint64)
+
+	require.NoError(t, err)
+	require.NoError(t, tenv.Env.ExistingAddresses.Merge(output.AddressBook))
+	// Get new state after migration.
+	state, err := changeset.LoadOnchainState(e)
+	require.NoError(t, err)
+
+	srcToken, dstToken := setupTokens(t, tenv)
+
+	// Ensure capreg logs are up to date.
+	changeset.ReplayLogs(t, e.Offchain, tenv.ReplayBlocks)
+
+	// Add all lanes
+	require.NoError(t, changeset.AddLanesForAll(e, state))
+	// Need to keep track of the block number for each chain so that event subscription can be done from that block.
+	startBlocks := make(map[uint64]*uint64)
+	// Send a message from each chain to every other chain.
+
+	// Mint 2 tokens to be transferred
+	twoCoins := new(big.Int).Mul(big.NewInt(1e18), big.NewInt(2))
+
+	linkToken := state.Chains[tenv.HomeChainSel].LinkToken
+	maxUint256 := math.MaxBig256
+
+	emptyTokenArray := make([]router.ClientEVMTokenAmount, 0)
+
+	// Create two ClientEVMTokenAmount structs to be passed to the router
+	tokens := map[uint64][]router.ClientEVMTokenAmount{
+		tenv.HomeChainSel: {{
+			Token:  srcToken.Address(),
+			Amount: twoCoins,
+		}},
+		tenv.FeedChainSel: {{
+			Token:  dstToken.Address(),
+			Amount: twoCoins,
+		}},
+	}
+
+	t.Run("Send message Pay with Link token home chain -> remote", func(t *testing.T) {
```

Comment:
As our intention is to test fee token, I think we can avoid testing this is in different lanes. We can test it in the same lane with different token by that we can simplify the test more.

Reply:
The original ticket specified that each test scenario should flip the lane direction

Reply:
I feel it is not needed. @makramkd @mateusz-sekara What do you guys think about this?